src/components/Contacts/ContactDetails/ContactDetails.test.tsx

```diff
@@ -1,44 +1,170 @@
-import React from 'react';
+import React, { useContext, useEffect, useState } from 'react';
 import { ThemeProvider } from '@mui/material/styles';
-import { render } from '@testing-library/react';
+import { act, render, waitFor } from '@testing-library/react';
 import userEvent from '@testing-library/user-event';
 import { SnackbarProvider } from 'notistack';
 import TestRouter from '__tests__/util/TestRouter';
 import { GqlMockedProvider } from '__tests__/util/graphqlMocking';
 import { ContactsWrapper } from 'pages/accountLists/[accountListId]/contacts/ContactsWrapper';
 import theme from '../../../theme';
-import { ContactDetailProvider } from './ContactDetailContext';
+import {
+  ContactDetailContext,
+  ContactDetailProvider,
+} from './ContactDetailContext';
 import { ContactDetails } from './ContactDetails';
 
 const accountListId = 'account-list-1';
 const contactId = 'contact-1';
 
-const router = {
-  query: { accountListId, contactId: [contactId] },
+const createRouter = (
+  overrides: Partial<Record<string, string | string[]>> = {},
+) => {
+  return {
+    pathname: '/contacts',
+    query: {
+      accountListId,
+      contactId: [contactId],
+      ...overrides,
+    },
+    replace: jest.fn(),
+    isReady: true,
+  } as {
+    pathname: string;
+    query: Record<string, string | string[] | undefined>;
+    replace: jest.Mock;
+    isReady: boolean;
+  };
+};
+
+const renderWithProviders = (
+  router: ReturnType<typeof createRouter>,
+  Component: React.FC,
+) => {
+  return render(
```

Comment:
you can still return the `render`


```suggestion
) => render(
```